### PR TITLE
Replication Status

### PR DESCRIPTION
**Pull request by aluzzardi**
Add replication status to `docker info`:

```
$ docker info
Containers: 2
Images: 50
Storage Driver:
Role: primary
Strategy: spread
```

```
$ docker info
Containers: 2
Images: 50
Storage Driver:
Role: replica
Primary: 127.0.0.1:4000
Strategy: spread
```

Also, `--leader-election` is now `--replication`.

Cleaned up the router/proxy as well (now named `api.Primary` and `api.Replica`).

### CODE REVIEW

**Comment by aluzzardi** [comment]
ping @abronan @vieux

**Comment by aluzzardi** [comment]
Fixes #890

File: api/replica.go [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+	for _, route := range localRoutes {
+		if strings.HasSuffix(r.URL.Path, route) {
+			p.handler.ServeHTTP(w, r)
+			return
+		}
+	}
+
+	// Otherwise, forward.
+	if p.primary == "" {
+		httpError(w, "No cluster leader", http.StatusInternalServerError)
```
Comment (abronan) [inline api/replica.go]: Should we change the logs to mention **primary** instead of **leader**?

File: api/replica.go [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+		return
+	}
+
+	if err := hijack(p.tlsConfig, p.primary, w, r); err != nil {
+		httpError(w, fmt.Sprintf("Unable to reach cluster leader: %v", err), http.StatusInternalServerError)
```
Comment (abronan) [inline api/replica.go]: same

File: cli/manage.go [hunk 3]
```diff
@@ -123,12 +147,14 @@ func setupLeaderElection(server *api.Server, apiHandler http.Handler, discovery
 		for leader := range leaderCh {
 			log.Infof("New leader elected: %s", leader)
```
Comment (abronan) [inline cli/manage.go]: same here, does `New primary elected` makes sense?

**Comment by abronan** [comment]
`leadership` Candidate test continues to fail (randomly on other PRs and this one).

Other than that and my comments, LGTM.

**Comment by aluzzardi** [comment]
PTAL

**Pull request merged**